### PR TITLE
US-2071 EOA Wallet - Confirmed TX gas amounts are incorrect

### PR DESCRIPTION
**Pull request by TravellerOnTheRun**
https://github.com/rsksmart/rif-wallet/assets/47615361/36600a32-f4ae-4c5a-8ebb-263f410bcdad

Simulator Screenshot - iPhone 15 - 2024-01-10 at 12 09 47

I changed the `activityDeserializer` to only consider `gasUsed` if none use 0, for example when transaction is pending.

### CODE REVIEW

**Review by jormelCoin — CHANGES_REQUESTED** [review]
The fee shown is not the same what appears in Explorer.

Screenshot 2024-01-10 at 14 07 28

Screenshot 2024-01-10 at 14 07 55

@jessgusclark

**Review by jessgusclark — CHANGES_REQUESTED** [review]
Sorry this should have been in the ticket but @jormelCoin is correct. Gas amounts should be displayed in the same format as the explorer shows. So, in this case the gas fee should say: 36,123 instead of 0.000000....

The logic could be the following, when the fee is in RBTC, then it should be displayed as 36,xxx, but when the fee is an erc20 token (like RIF), then we need to take the value, get the decimal places and convert to something like 2.136.

**Comment by TravellerOnTheRun** [comment]
> Sorry this should have been in the ticket but @jormelCoin is correct. Gas amounts should be displayed in the same format as the explorer shows. So, in this case the gas fee should say: 36,123 instead of 0.000000....
> 
> The logic could be the following, when the fee is in RBTC, then it should be displayed as 36,xxx, but when the fee is an erc20 token (like RIF), then we need to take the value, get the decimal places and convert to something like 2.136.

@jormelCoin @jessgusclark Please, check again, did it like you asked.

**Comment by jormelCoin** [comment]
@TravellerOnTheRun, I get this message when trying to test this pr.

https://github.com/rsksmart/rif-wallet/assets/114427184/e2066921-6d26-47f8-8915-6afdae2a3f4f

**Review by jormelCoin — CHANGES_REQUESTED** [review]
Hey @TravellerOnTheRun, there is a slight difference, so you're missing the comma( , )

Screenshot 2024-01-16 at 11 53 41

Screenshot 2024-01-16 at 11 53 57

At the same time, I'd like to know if it should display the conversion below. I guess so.
cc @jessgusclark

**Comment by jormelCoin** [comment]
In old transactions, if it shows the requested format. However, when making a shipment, it shows all 000000.
https://github.com/rsksmart/rif-wallet/assets/114427184/f07349dc-31a5-4b84-9feb-ac6ef68505b6

**Review by jormelCoin — CHANGES_REQUESTED** [review]
As we discussed in the daily, this screen should have the same format.

Screenshot 2024-01-17 at 13 02 47

**Comment by jormelCoin** [comment]
@TravellerOnTheRun Please keep in mind the conversion since I'm not seeing it in the transaction list.

**Comment by TravellerOnTheRun** [comment]
@jessgusclark Can you run tests locally? it throws bullshit errors like TokenSymbol being undefined, I've looked through the files it's defined everywhere. Locally for me it argues that there's no 'castStyle.view' in 6 files, WTF?! :D

**Comment by jessgusclark** [comment]
@TravellerOnTheRun Yes, I am getting the `castStyle.view` error and `TRBTC` errors when I run it locally. I don't understand why this is now though, as it looks like it was passing before just fine. 

Screenshot 2024-01-19 at 2 30 11 PM

**Comment by TravellerOnTheRun** [comment]
> @TravellerOnTheRun Yes, I am getting the `castStyle.view` error and `TRBTC` errors when I run it locally. I don't understand why this is now though, as it looks like it was passing before just fine.
> 
> Screenshot 2024-01-19 at 2 30 11 PM

It looks like a malfunction, what can go wrong? It's there it's a fact, it does not fail when you work on the app, every screen has `castStyle.view` it would show the error.

**Comment by jessgusclark** [comment]
@TravellerOnTheRun If I go back one commit, the tests work, `42a58103d721d`. So something in the last commit broke it.

**Comment by TravellerOnTheRun** [comment]
> @TravellerOnTheRun Yes, I am getting the `castStyle.view` error and `TRBTC` errors when I run it locally. I don't understand why this is now though, as it looks like it was passing before just fine.
> 
> Screenshot 2024-01-19 at 2 30 11 PM

I also thing is some sort of malfunction of the test runner because locally it's one error, here it's completely different

File: src/shared/utils/index.ts [hunk 1]
```diff
@@ -21,9 +21,51 @@ import { RifRelayConfig } from '@rsksmart/rif-relay-light-sdk'
...
+import { TokenSymbol } from 'screens/home/TokenImage'
+
 import { Wallet } from '../wallet'
 import { ErrorWithMessage } from '../types'
 
+export const rbtcMap = new Map([
+  [TokenSymbol.TRBTC, true],
+  [TokenSymbol.RBTC, true],
+  [undefined, false],
+])
```
Comment (jessgusclark) [inline src/shared/utils/index.ts]: Looks like this was moved, right? So perhaps there was a reference to it that is now missing?

File: src/shared/utils/index.ts [hunk 2]
```diff
@@ -21,9 +21,51 @@ import { RifRelayConfig } from '@rsksmart/rif-relay-light-sdk'
...
+  }
+
+  const longAssNumberArr = longAssNumber.split('')
+
+  for (let i = longAssNumber.length - 3; i > 0; i -= 3) {
+    longAssNumberArr.splice(i, 0, ',')
+  }
+
+  return longAssNumberArr.join('')
+}
```
Comment (jessgusclark) [inline src/shared/utils/index.ts]: Rename these to remove `ass`.

**Review by jessgusclark — CHANGES_REQUESTED** [review]

File: src/shared/utils/index.ts (hunk 1 above)
Comment (TravellerOnTheRun) [reply src/shared/utils/index.ts]: Checked by the name, it's there

**Comment by jormelCoin** [comment]
It now works as expected. However, when sending BTC, I get a NaN in the conversion. So, Should I create a ticket for it, or would rather work on this PR?
Screenshot 2024-01-23 at 09 54 30

@TravellerOnTheRun

**Comment by TravellerOnTheRun** [comment]
> It now works as expected. However, when sending BTC, I get a NaN in the conversion. So, Should I create a ticket for it, or would rather work on this PR?
> 
> Screenshot 2024-01-23 at 09 54 30
> 
> 
> 
> @TravellerOnTheRun 
> 
> 

Bitcoin gas is not in RBTC so yes, it should be in another ticket. You can create a ticket for EOA Wallet.

**Review by jormelCoin — APPROVED** [review]
This ticket works as expected ✅

**Pull request merged**